### DISCUSSION TITLE
Bad if statement?

### DISCUSSION THREAD

**Discussion by Glort572** [ohmybash/oh-my-bash#475] [Q&A]
Hey, I'm just curious. Shouldn't there be `==` instead of `=` in the if statement at line 21 of [powerline.base.sh](https://github.com/ohmybash/oh-my-bash/blob/4c7a0ebb8d11efe0b730cc7e5f9961428741f5cc/themes/powerline/powerline.base.sh#L21C36-L21C36)?

**Comment by akinomyoga** [comment]
Line 21 of which file are you talking about?

We want to normalize it from `=` to `==`, but both actually works practically. The POSIX standard only defines `=` for the comparison of strings in the commands `[` / `test`. On the other hand, ksh `[[ ... ]]` has many differences from the POSIX standard `[ ... ]`. The choice of operator from `==` and `=` is one of them. Although `==` is not standardized, it is widely used in `[[ ... ]]` in ksh/bash/zsh/etc.

**Reply by Glort572** [reply]
I was talking about this file: https://github.com/ohmybash/oh-my-bash/blob/4c7a0ebb8d11efe0b730cc7e5f9961428741f5cc/themes/powerline/powerline.base.sh#L21C36-L21C36

**Reply by Glort572** [reply]
By the way, thanks for answering and clearing my doubt! :)
